users/urls.py

```diff
@@ -15,6 +21,7 @@
     url(r'^logout/$', views.UserLogoutView, name='logout'),
     url(r'^update-password/$', views.UserPasswordView, name='update-password'),
     url(r'^(?P<pk>\d+)/$', UserProfileView.as_view(), name='profile'),
+    url(r'^(?P<username>\w+)/$', views.UserProfileViewByUsername, name='profile_username'),
```

Comment:
that could be fixed really fast, by just validating that the usernames must be at least any size of characters and not begin with a number, as everywhere else.